### ISSUE TITLE
Alexey-Ben meeting 2018-11-05

### ISSUE THREAD

**Issue by ashiklom** [FoRTExperiment/FoRTE-mgmt#18]
Goals from last week:

- **ED bare ground runs** -- Figured out how to get ED running, but haven't started runs yet.
    - Best way I can see is to change some hard-coded limits in the source code. Because it requires source code changes, I took some time to set up a new "develop" version of ED in PEcAn.
    - Also, found a new ED/PEcAn bug related to PFTs. Need to fix how PEcAn tells ED which PFTs to run. I know how, but haven't done it yet.
- **ED RTM R implementation** -- Successfully debugged and have it working! Next step is to use it for parameter sensitivity analysis.
- **FoRTE paper** -- Added some more useful notes and references from my general reading, but overall not much progress.
- **PEcAn** -- Spent some time reconciling my local changes against new changes to PEcAn mainline. I need to be better about not branching too far off the PEcAn mainline. Fortunately, figured out how to easily override standard (mainline) Docker configuration with local changes, so should be easier to stay in sync with main PEcAn code moving forward.
- **Better SIPNET runs** -- Didn't get to.
- **Background reading on SIPNET** -- Didn't get to.
- **Hector** -- Didn't get to.
- **Other**
    - Review a paper revision for AgForMet
    - Help with job interviews

Goals for this week:

- FoRTE
    - ED-related fixes to PEcAn
    - Natively pass arguments to ED binary through PEcAn. This is needed to fix [#2146](https://github.com/PecanProject/pecan/issues/2146).
    - Run ED only with PFTs specified in PEcAn configuration. Currently, defaults to all ED PFTs for historical reasons, but this causes a nasty bug with new liana PFT, and is generally bad for bare ground runs. (See [#2178](https://github.com/PecanProject/pecan/issues/2178)).
    - ED bare ground runs at UMBS, mostly to make sure they run
    - (Time-permitting) Revisit SiPNET runs.
- ED RTM (AGU presentation)
    - Implement very simple soil moisture-reflectance model (just the weighted mean of wet and dry soil spectra). I need this to account for very different soil reflectance across different sites.
    - Implement ED leaf and wood allometries in R. ED cannot take LAI or WAI as input; only DBH, such that leaf biomass = f(DBH|leaf/wood allometry params) and LAI = leaf biomass * SLA. Something similar for wood.
    - Sensitivity analysis:
        - LAI (overall, and distribution)
        - Wood area index (WAI)
        - Leaf and wood reflectance and transmittance
        - Soil reflectance
        - Leaf angle distribution and clumping factor
        - Fraction of direct vs. diffuse incident radiation
    - (Time-permitting) Calibration
- Background reading -- RSS feeds for journals, Google Scholar alerts, older papers on my to-read list. Currently, takes ~1 hour per day. Is this too much?

**Comment by bpbond** [comment]
>Background reading

I don't think this is too much, although I wouldn't let it go higher (in general; obviously if doing reading for a paper you're writing this can spike up).

What about that liana divide-by-zero problem? Solution proposed in https://github.com/PecanProject/pecan/issues/2178 doesn't mention this. Will a fix get upstreamed to ED? Just wondering.

**Comment by ashiklom** [comment]
> although I wouldn't let it go higher

Sounds good.

> What about that liana divide-by-zero problem?

There is someone visiting Mike's lab that said he is aware of this and actively working on it (or, at least, plans to work on it). I'll try to get him to open an issue about it in the ED2 repo, since he will know more about the underlying cause.
